### PR TITLE
Alexmont/aggregate adapter

### PR DESCRIPTION
**Pull request by alexmontAmazon**
## What does this PR do?
- Implement first working version of DPE-based multi-edit via a new RowAggregateAdapter class
- create GHIs for missing features
- create a GenericButton parent for ContainerActionButton so the DPE can have generic buttons in handlers
- put aztoolsframework_files.cmake back into alphabetical order

## How was this PR tested?
locally using the DPEDebugViewStandalone app

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h [hunk 1]
```diff
@@ -18,19 +18,19 @@ namespace AZ::DocumentPropertyEditor
         ValueStringFilter();
         void SetFilterString(AZStd::string filterString);
 
+    protected:
+        // pure virtual overrides
+        virtual MatchInfoNode* NewMatchInfoNode() const override;
+        virtual void CacheDomInfoForNode(const Dom::Value& domValue, MatchInfoNode* matchNode) const override;
+        virtual bool MatchesFilter(MatchInfoNode* matchNode) const override;
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h:25]: The `virtual` keyword should be removed.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/GenericButtonHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            });
+    }
+
+    void GenericButtonHandler::SetValueFromDom(const AZ::Dom::Value& node)
+    {
+        using AZ::DocumentPropertyEditor::Nodes::GenericButton;
+        // Cache the node so we can query OnActivate on it when we're pressed.
+        m_node = node;
+        auto buttonText = GenericButton::ButtonText.ExtractFromDomNode(node).value_or("");
+        setText(QString::fromLatin1(AZStd::string(buttonText).c_str()));
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/GenericButtonHandler.cpp:32]: This should use `QString::fromUtf8` not the latin1 encoding as that doesn't support unicode.
```suggestion
        setText(QString::fromUtf8(AZStd::string(buttonText).c_str()));
```

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AggregateAdapter.h [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        // monotonically increasing frame counter that increments whenever a source adapter gets an update
+        unsigned int m_updateFrame = 0;
+
+        AdapterBuilder m_builder;
+    };
+
+    class LabeledRowAggregateAdapter : public RowAggregateAdapter
+    {
+    public:
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AggregateAdapter.h:109]: The public section isn't needed.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h (hunk 1 above)
Comment (alexmontAmazon) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h]: done

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/GenericButtonHandler.cpp (hunk 2 above)
Comment (alexmontAmazon) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/GenericButtonHandler.cpp]: yep, done thanks

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AggregateAdapter.h (hunk 3 above)
Comment (alexmontAmazon) [reply Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AggregateAdapter.h]: removed.

**Review by amzn-tmryan — APPROVED** [review]

**Review by dtamkin1 — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h [hunk 4]
```diff
@@ -18,19 +18,19 @@ namespace AZ::DocumentPropertyEditor
         ValueStringFilter();
         void SetFilterString(AZStd::string filterString);
 
+    protected:
+        // pure virtual overrides
```
Comment (lsemp3d) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ValueStringFilter.h:22]: minor: perhaps change the comment to say where the overrides are from, `RowFilterAdapter`

**Review by lsemp3d — APPROVED** [review]

**Pull request merged**